src/content/app/genome-browser/state/track-config/trackConfigSlice.ts

```diff
@@ -21,43 +21,73 @@ import {
   PayloadAction,
   ThunkAction
 } from '@reduxjs/toolkit';
+import { RootState } from 'src/store';
 
 import browserStorageService from 'src/content/app/genome-browser/services/browserStorageService';
+import browserTrackConfigStorageService from '../../components/browser-track-config/services/browser-track-config-storage-service';
+import { BrowserTrackStates } from 'src/content/app/genome-browser/components/track-panel/trackPanelConfig';
 
 import { updateTrackStates } from 'src/content/app/genome-browser/state/browser-general/browserGeneralSlice';
 import { getBrowserTrackStates } from 'src/content/app/genome-browser/state/browser-general/browserGeneralSelectors';
 
-import { BrowserTrackStates } from 'src/content/app/genome-browser/components/track-panel/trackPanelConfig';
-import { RootState } from 'src/store';
+import { GenomeTrackCategory } from 'src/shared/state/genome/genomeTypes';
 
 export type CogList = {
   [key: string]: number;
 };
 
-export type TrackConfigState = Readonly<{
+export enum TrackType {
+  GENE = 'gene',
+  REGULAR = 'regular'
+}
+
+export const getTrackType = (trackId: string) => {
+  if (trackId.startsWith('gene')) {
+    return TrackType.GENE;
+  } else {
+    return TrackType.REGULAR;
+  }
+};
+
+export type GeneTrackConfig = {
+  showMoreTranscripts: boolean;
+  showTranscriptIds: boolean;
+  showTrackName: boolean;
+  showFeatureLabel: boolean;
+  trackType: TrackType.GENE;
+};
+
+export type RegularTrackConfig = {
+  showTrackName: boolean;
+  trackType: TrackType.REGULAR;
+};
+
+export type TrackConfig = GeneTrackConfig | RegularTrackConfig;
+
+export type TrackInfo = {
+  [trackId: string]: TrackConfig;
+};
+
+export type TrackConfigStateForObject = Readonly<{
   applyToAllConfig: {
-    isSelected: boolean;
-    allTrackNamesOn: boolean;
-    allTrackLabelsOn: boolean;
+    isSelected?: boolean;
   };
-  browserCogList: number;
-  browserCogTrackList: CogList;
+  browserCogList: CogList;
   selectedCog: string | null;
-  trackConfigNames: { [key: string]: boolean };
-  trackConfigLabel: { [key: string]: boolean };
+  tracks: TrackInfo;
```

Comment:
The name `TrackInfo` reads as if it means is some info about some track; but the way you declare it, it looks like a map of track id to track config. We need a better name for this. `TrackConfigs`? `TrackConfigsMap`?

Reply:
Ok, will take `TrackConfigs`